### PR TITLE
Issue #11637: Use Shellcheck to resolve SC2006 in Shell Script

### PR DESCRIPTION
**Pull request by Rahulkhinchi03**
Resolves `SC2006` : https://github.com/checkstyle/checkstyle/issues/11637

SC2006: (style): Use $(...) notation instead of legacy backticks `...`.

### CODE REVIEW

**Comment by Rahulkhinchi03** [comment]
https://github.com/checkstyle/checkstyle/blob/7c5c543cb1bdd3a72de806f57560c9b343508b67/.travis.yml#L75-L78

@romani @nick-mancuso @rnveach As we have given the directory to check only `./.ci` files, Shellcheck doesn't check errors in `release.sh` as it is in the root directory and not in `.ci`

Well, `release.sh` has only 2 ERROR CODE: `SC2006 `and `SC2086`, So I am making a minor commit to resolve `SC2086` (Because it is already resolved in `.ci` folder scripts): https://github.com/checkstyle/checkstyle/pull/11650

**Comment by romani** [comment]
`shellcheck ./.ci/*.sh && shellcheck release.sh`

**Comment by Rahulkhinchi03** [comment]
So as of now, Should I just do:
`- CMD="shellcheck ./.ci/*.sh && shellcheck release.sh" `

**Comment by Rahulkhinchi03** [comment]
@nick-mancuso I didn't find any error with this code but I found a lot of others. 

I believe I should create a new issue for yml files and add those errors in `.shellcheckrc`

**Comment by nick-mancuso** [comment]
```
➜  checkstyle git:(master) grep -IR "=\`" | grep yml
.github/workflows/site.yml:          TIME=`date +%Y%H%M%S`
.github/workflows/diff_report.yml:        TIME=`date +%Y%H%M%S`


```

**Comment by Rahulkhinchi03** [comment]
> 

@nick-mancuso  What about yml files? 

`shellcheck ./.ci/*.sh && shellcheck release.sh && shellcheck ./*.yml`

File: .ci/validation.sh [hunk 1]
```diff
@@ -438,15 +438,15 @@ check-since-version)
...
   if git show --summary HEAD | grep ^Merge: ; then
       echo "Merge detected."
-      HEAD=`git log -n 1 --no-merges --pretty=format:"%H"`
+      HEAD=$(git log -n 1 --no-merges --pretty=format:"%H")
   fi
   # Identify previous commit to know how much to examine
   # Script assumes we are only working with 1 commit if we are in master
   # Otherwise, it looks for the common ancestor with master
-  COMMIT=`git rev-parse "$HEAD"`
+   COMMIT=$(git rev-parse "$HEAD")
```
Comment (nick-mancuso) [inline .ci/validation.sh:449]: Please remove preceding whitespace

**Review by nick-mancuso — CHANGES_REQUESTED** [review]
Item:

File: .ci/validation.sh (hunk 1 above)
Comment (nick-mancuso) [reply .ci/validation.sh]: @Rahulkhinchi03 please carefully look through your commits, removal is not in same commit as update

Comment (Rahulkhinchi03) [reply .ci/validation.sh]: Changes Done

**Review by romani — APPROVED** [review]

**Pull request merged**